### PR TITLE
feat(callingsdk): initial changes to add bnr

### PR DESCRIPTION
**Pull request by Kesari3008**
# COMPLETES #[SPARK-433414](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-433414)

## This pull request addresses
Adding changes to use the BNR feature in CallingSDK for WebRTC Calling.

## by making the following changes
Added static function to create and return NoiseReductionEffect object in calling.js
Added the effect to the localMicrophoneStream being used for calls in app.js and enabled it.
Listening for 'output_track_change' event to update the tracks once BNR is enabled/disabled.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by Shreyas281299 — COMMENTED** [review]
The changes look good. Just one minor comment

File: packages/calling/src/CallingClient/calling/call.ts [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
-import {Event, LocalMicrophoneStream, RoapMediaConnection} from '@webex/internal-media-core';
+import {
+  Event,
+  LocalMicrophoneStream,
+  LocalStreamEventNames,
+  RoapMediaConnection,
+} from '@webex/internal-media-core';
 import {createMachine, interpret} from 'xstate';
 import {v4 as uuid} from 'uuid';
+import {LocalMicrophoneStreamEventNames} from '@webex/media-helpers';
```
Comment (Shreyas281299) [inline packages/calling/src/CallingClient/calling/call.ts:10]: Can we remove this import? This has not been used anywhere

Comment (Kesari3008) [reply packages/calling/src/CallingClient/calling/call.ts:10]: Removed it

**Review by Shreyas281299 — APPROVED** [review]

File: packages/calling/src/index.ts [hunk 2]
```diff
@@ -9,9 +9,10 @@ import Logger from './Logger';
...
   createClient,
   createCallHistoryClient,
+  createCallSettingsClient,
+  createContactsClient,
   createMicrophoneStream,
   createVoicemailClient,
-  createContactsClient,
-  createCallSettingsClient,
   Logger,
+  NoiseReductionEffect,
```
Comment (BhargavSatya) [inline packages/calling/src/index.ts:17]: Do we have to export `NoiseReductionEffect` what is its use exporting?

**Review by BhargavSatya — APPROVED** [review]
Approved With a minor comment.

File: packages/calling/src/index.ts (hunk 2 above)
Comment (Kesari3008) [reply packages/calling/src/index.ts:17]: If we don't export, I cannot use it in calling.js

Comment (BhargavSatya) [reply packages/calling/src/index.ts:17]: I thought developers will use this method to create the NoiseReductionEffect:   static createNoiseReductionEffect(authToken) . What is this about?

Comment (Kesari3008) [reply packages/calling/src/index.ts:17]: Yeah, that's what developers are using. But did you see what's happening inside that static function? We are using NoiseReductionEffect inside that function. and to get access of the same, it needs to be re-exported from index.ts

Comment (BhargavSatya) [reply packages/calling/src/index.ts:17]: I did see. Just noticed the it was a js file. Lack of any return type in the signature and description confused me. Nevermind.

**Pull request merged**